File: conformance/bundles/14-stateless-steps/SquaresSteps.java
package dev.varar.conformance.bundle14;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;
import java.util.List;

/**
 * Java sibling of {@code squares.steps.ts} / {@code squares.steps.py} /
 * {@code squares.steps.kt} (bundle {@code 14-stateless-steps}): no state factory —
 * these steps are pure, so the factory-less {@code steps()} binds handlers to
 * {@link State.Empty}.
 */
public final class SquaresSteps implements StepDefinitions<State.Empty> {

    @Override
    public void register(Steps<State.Empty> s) {

        s.stimulus("I warm up my mental math", (State.Empty state) -> state);

        s.sensor(
                "The square of {int} is {int}.",
                (State.Empty state, Integer n, Integer expected) -> List.of(n, n * n));
    }
}

File: conformance/bundles/14-stateless-steps/squares.steps.cs
// C# sibling of squares.steps.ts / .rs (bundle 14-stateless-steps).
// Pure steps — nothing to arrange or evolve — so State() is a bare Null every handler ignores.
using Varar;
using Varar.Core;

namespace Varar.Corpus.B14;

public static class SquaresSteps
{
    public static void Register(Steps s)
    {
        s.Stimulus("I warm up my mental math", state => null);

        // Two slots ({int}, {int}); the handler uses only the first and returns both computed
        // columns [n, n*n] for positional comparison.
        s.Sensor("The square of {int} is {int}.", (state, n, square) =>
        {
            var value = n.AsInt();
            return Value.List([Value.Of(value), Value.Of(value * value)]);
        });
    }

    public static Value State() => Value.Null;
}
